cmake/sysbuild/fast_pair_hex.cmake

```diff
@@ -47,6 +47,36 @@ function(fast_pair_hex_pm)
     )
 endfunction()
 
+# This function calculates the absolute partition address by summing up
+# all hex numbers preceded by @ in the `partition_node_full_path` string.
+# The result is stored in the variable provided as `absolute_addr`.
+# This function should probably one day be moved to some common utilities file.
+function(get_partition_absolute_addr absolute_addr partition_node_full_path)
+  set(addr_regex_pattern "@([0-9a-fA-F]+)($|/)")
```

Comment:
this is just... no. We have dt functions for this sort of thing. Add this function:
```
function(dt_get_parent node)
  string(FIND "${${node}}" "/" pos REVERSE)

  if(pos EQUAL -1)
    message(FATAL_ERROR "Unable to get parent of node: ${${node}}")
  endif()

  string(SUBSTRING "${${node}}" 0 ${pos} ${node})
  set(${node} "${${node}}" PARENT_SCOPE)
endfunction()
```
Then get the address of the node, get the 2x parent, get that address **using the proper DT functions**, add them, and you have your result
```
sysbuild_dt_reg_addr(addr1 IMAGE ${DEFAULT_IMAGE} PATH ${bt_fast_pair_partition_node_full_path})
dt_get_parent(bt_fast_pair_partition_node_full_path)
dt_get_parent(bt_fast_pair_partition_node_full_path)
sysbuild_dt_reg_addr(addr2 IMAGE ${DEFAULT_IMAGE} PATH ${bt_fast_pair_partition_node_full_path})
math(EXPR combined "${addr1} + ${addr2}" OUTPUT_FORMAT HEXADECIMAL)
message(WARNING "is ${combined}")
```
=
> is 0xe1e8fb8

Reply:
You still need to parse string to get a parent so I thought I can parse it and just add addresses. Anyway, I will do as you suggest. @tejlmand, do you approve Jamie's suggestion?